src/testRunner/unittests/config/commandLineParsing.ts

```diff
@@ -414,90 +414,181 @@ namespace ts {
                 });
         });
 
-        describe("parse --composite", () => {
-            it("allows setting it to false", () => {
-                assertParseResult(["--composite", "false", "0.ts"],
-                    {
-                        errors: [],
-                        fileNames: ["0.ts"],
-                        options: { composite: false }
-                    });
-            });
-
-            it("allows setting it to undefined and tsbuild info to undefined", () => {
-                assertParseResult(["--composite", "undefined", "-tsBuildInfoFile", "undefined", "0.ts"],
-                    {
-                        errors: [],
-                        fileNames: ["0.ts"],
-                        options: { composite: undefined, tsBuildInfoFile: undefined }
-                    });
-            });
-
-            it("allows setting it to null and tsbuild info to null", () => {
-                assertParseResult(["--composite", "null", "-tsBuildInfoFile", "null", "0.ts"],
-                    {
-                        errors: [],
-                        fileNames: ["0.ts"],
-                        options: { composite: undefined, tsBuildInfoFile: undefined }
+        describe("parses command line null for tsconfig only option", () => {
+            interface VerifyNull {
+                optionName: string;
+                nonNullValue?: string;
+                workerDiagnostic?: () => ParseCommandLineWorkerDiagnostics;
+                diagnosticMessage: DiagnosticMessage;
+            }
+            function verifyNull({ optionName, nonNullValue, workerDiagnostic, diagnosticMessage }: VerifyNull) {
+                it("allows setting it to null", () => {
+                    assertParseResult(
+                        [`--${optionName}`, "null", "0.ts"],
+                        {
+                            errors: [],
+                            fileNames: ["0.ts"],
+                            options: { optionName: undefined }

```

Comment:
```suggestion
                            options: { [optionName]: undefined }

```

Reply:
... how was this passing, actually

Reply:
oh, hah, because since `optionName` never existed, it was always `undefined`. Eeeeyeah.